IdentityCore/src/cache/accessor/MSIDDefaultTokenCacheAccessor.m

```diff
@@ -973,6 +1036,33 @@ - (BOOL)saveAccount:(MSIDAccount *)account
     return [NSSet setWithArray:[refreshTokens valueForKey:@"homeAccountId"]];
 }
 
+- (void)fillIdTokenClaimsForAccounts:(NSArray<MSIDAccount *> *)accounts
+                        withIdTokens:(NSArray<MSIDIdToken *> *)idTokens
+{
+    NSMutableDictionary *searchMap = [NSMutableDictionary new];
+    for (MSIDIdToken *idToken in idTokens)
+    {
+        NSString *key = [NSString stringWithFormat:@"%@-%@-%@", idToken.accountIdentifier.homeAccountId, idToken.authority.environment, idToken.authority.url.msidTenant];
+        [searchMap setValue:idToken forKey:key];
+    }
+    
+    for (MSIDAccount *account in accounts)
```

Comment:
nit: we are already looping through accounts once above. Do we want to combine id token setting with that loop?

Reply:
addressed by moving it into that loop :)